helix-vcs/src/git.rs

```diff
@@ -51,6 +99,111 @@ impl Git {
         )
         .ok()
     }
+
+    /// Emulates the result of running `git status` from the command line.
+    fn status(repo: &Repository) -> Result<Vec<FileChange>> {
+        let autocrlf = repo
+            .config_snapshot()
+            .boolean("core.autocrlf")
+            .unwrap_or(false);
+
+        let work_dir = repo
+            .work_dir()
+            .ok_or_else(|| anyhow::anyhow!("working tree not found"))?;
+
+        // TODO: allow diffing against another ref
+        let head_tree = repo.head_commit()?.tree()?;
+        let head_state = State::from_tree(&head_tree.id, |oid, buf| {
+            repo.objects.find_tree_iter(oid, buf).ok()
+        })?;
+
+        let mut head_tree_set = HashSet::new();
+        let mut submodule_paths = vec![];
+
+        let mut raw_changes = RawChanges::default();
+
+        for item in head_state.entries() {
+            let full_path = work_dir.join(&PathBuf::from(item.path(&head_state).to_string()));
+
+            if item.mode == Mode::COMMIT {
+                submodule_paths.push(full_path);
+            } else {
```

Comment:
Removes an indentation level, but for this one I'm not sure it's better

```suggestion
            if item.mode == Mode::COMMIT {
                submodule_paths.push(full_path);
                continue;
            }
            
            // ...
```